digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -260,123 +250,103 @@ trait LaneOperations {
             (missingLanes, Seq())
 
-
           // case both main lanes exists
           case _ =>
-            val needAdjustSideCodeLane11 = getLaneCodesForSideCodeAdjust(lanesToProcess, MainLane.towardsDirection, FourthRightAdditional.towardsDirection, mainLane11SideCode)
-            val needAdjustSideCodeLane21 = getLaneCodesForSideCodeAdjust(lanesToProcess, MainLane.againstDirection, FourthRightAdditional.againstDirection, mainLane21SideCode)
+            val adjustedSideCode11 = editLanesNeedSideCodeAdjust(lanesToProcess, MainLane.towardsDirection, FourthRightAdditional.towardsDirection, mainLane11SideCode)
+            val adjustedSideCode21 = editLanesNeedSideCodeAdjust(lanesToProcess, MainLane.againstDirection, FourthRightAdditional.againstDirection, mainLane21SideCode)
 
-            (Seq(), needAdjustSideCodeLane11 ++ needAdjustSideCodeLane21)
+            (Seq(), adjustedSideCode11 ++ adjustedSideCode21)
         }
 
-        // To remove the lanes with wrong SideCode
-        val toRemove = lanesToProcess.filter { lane =>
-                                        val isValidLaneId = lane.id != 0L
-                                        val isLaneCodeTowards = lane.laneCode >= MainLane.towardsDirection && lane.laneCode <= FourthRightAdditional.towardsDirection
-                                        val isSideCodeTowardsWrong = lane.sideCode != mainLane11SideCode.value && lane.sideCode != SideCode.BothDirections.value
-
-                                        val isLaneCodeAgainst = lane.laneCode >= MainLane.againstDirection && lane.laneCode <= FourthRightAdditional.againstDirection
-                                        val isSideCodeAgainstWrong = lane.sideCode != mainLane21SideCode.value && lane.sideCode != SideCode.BothDirections.value
-
-                                        val isLaneToRemove = (isLaneCodeTowards && isSideCodeTowardsWrong ) || (isLaneCodeAgainst && isSideCodeAgainstWrong)
-
-                                        isValidLaneId && isLaneToRemove
-                                     }
-                                    .map(_.id)
-
-        val lanesToAdd = (lanes ++ toAdd).filterNot(lane => toRemove.contains(lane.id))
+        val updatedLanes = (lanes ++ toAdd).filterNot(lane => adjustedSideCode.exists(_.id == lane.id)) ++ adjustedSideCode
 
-        val newChangeSet = changeSet.copy(generatedPersistedLanes = changeSet.generatedPersistedLanes ++ lanesToAdd,
-                                    expiredLaneIds = changeSet.expiredLaneIds ++ toRemove,
-                                    adjustedSideCodes = changeSet.adjustedSideCodes ++ needAdjustSideCode )
-
-        (lanesToAdd, newChangeSet)
+        val lanesWithSideCodeAdjustment = adjustedSideCode.map(lane => SideCodeAdjustment(lane.id, SideCode(lane.sideCode)))
+        val newChangeSet = changeSet.copy(generatedPersistedLanes = updatedLanes,
+          adjustedSideCodes = changeSet.adjustedSideCodes ++ lanesWithSideCodeAdjustment)
```

Comment:
Adding lanesWithSideCodeAdjustmentin the generatedPersistedLanes and in the adjustedSideCodes
Will fire 2 actores
1 to update the lane and another to expire the lane and create a new one.

What is supose to happen here?
remove and create a new one? Just update?

Reply:
In the end the generatedPersistedLanes only process the lanes with id = 0. The adjustedSideCode attribute of the changedSet will do the update(in the code removes and then add another one with the updated info)

Reply:
ok